src/main/java/mate/jdbc/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,133 @@
+            );
+        }
+        return Optional.ofNullable(manufacturer);
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        String query = "SELECT * "
+                     + "FROM manufacturers "
+                     + "WHERE is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllManufacturersStatement =
+                        connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturers.add(parseManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Select of all manufacturers from DB was failed.", e);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE manufacturers SET name = ?, country = ? "
+                     + "WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement = connection.prepareStatement(query)) {
+            getManufacturerStatement.setString(1, manufacturer.getName());
+            getManufacturerStatement.setString(2, manufacturer.getCountry());
+            getManufacturerStatement.setLong(3, manufacturer.getId());
+            getManufacturerStatement.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Update of manufacturer to DB was failed. "
+                    + "Manufacturer: " + manufacturer, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE manufacturers SET is_deleted = true "
+                     + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement = connection.prepareStatement(query)) {
+            getManufacturerStatement.setLong(1, id);
+            return getManufacturerStatement.executeUpdate() >= 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Delete of manufacturer from DB was failed. "
+                    + "Id: " + id, e);
+        }
+    }
+
+    private Manufacturer parseManufacturer(ResultSet resultSet) {
+        try {
+            Long id = resultSet.getObject("id", Long.class);
+            String name = resultSet.getString("name");
+            String country = resultSet.getString("country");
+            return new Manufacturer() {
+                {
+                    setId(id);
+                    setName(name);
+                    setCountry(country);
+                }
+            };
```

Comment:
```suggestion
            return new Manufacturer(id, name, country);
```
add a constructor with all arguments